climada/util/coordinates.py

```diff
@@ -1057,6 +1057,79 @@ def assign_coordinates(coords, coords_to_assign, distance="euclidean",
                 coords_to_assign, coords[not_assigned_idx_mask], threshold, **kwargs)
     return assigned_idx
 
+def assign_haz_centroids(coord_df, hazard, crs, distance='euclidean',
+                        threshold=NEAREST_NEIGHBOR_THRESHOLD,
+                        overwrite=True):
+    """Assign to each exposure's coordinate point its closest hazard's coordinate.
+    If disatances > threshold in points' distances, -1 is returned.
+    If raster hazard and coordinate point outside raster, -1 is returned.
+
+    Parameters
+    ----------
+    coord_df : pd.DataFrame
+        Dataframe with defined latitude/longitude column
```

Comment:
shouldn't that rather be a `GeoDataFrame`? because then the `crs` argument would become obsolete and the check for equality would make sense.

Reply:
Originally it was a geoDataFrame. But since impact object don't have a geodataframe by default I thought I thought I make the method more flexible and for the impact.assign_centroids() I don't create a GeoDataFrame first, but directly pass the coord_df since the crs information is already saved in the impact object.
The most common use case (also when using the function on its own) will still be a GeoDataFrame. 
I could change it back again to only allow GeoDataFrames as input, if you think this is better?

Reply:
I resolved it by passing an additonal test_crs argument, which is True by default. Only when it's called from exp.assign_centroids(), the crs match is checked outside of the u_coord.assign_gdf_centroids() function, because the exp.gdf does not require an active geometry column, without which the equal crs cannot be tested